src/Shared/PlatformNegotiation.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.Evaluation;
+
+namespace Microsoft.Build.Shared
+{
+    /// <summary>
+    /// This class contains only static methods, which are used in both the 
+    /// tasks and graph projects in order for two projects to negotiate which platform a projectreference
+    /// should be built as.
+    /// </summary>
+    static internal class PlatformNegotiation
```

Comment:
Nit (per https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0036)

```suggestion
    internal static class PlatformNegotiation
```